pkg/optimizely/client.go

```diff
@@ -170,27 +172,63 @@ func (c *OptlyClient) RemoveForcedVariation(experimentKey, userID string) (*Over
 // ActivateFeature activates a feature for a given user by getting the feature enabled status and all
 // associated variables
 func (c *OptlyClient) ActivateFeature(key string, uc entities.UserContext, disableTracking bool) (*Decision, error) {
-	enabled, variables, err := c.GetAllFeatureVariables(key, uc)
+	projectConfig, err := c.OptimizelyClient.ConfigManager.GetConfig()
 	if err != nil {
 		return &Decision{}, err
 	}
 
-	// HACK - Triggers impression events when applicable. This is not
-	// ideal since we're making TWO decisions for each feature now. TODO OASIS-5549
-	if !disableTracking {
-		_, tErr := c.IsFeatureEnabled(key, uc)
-		if tErr != nil {
-			return &Decision{}, tErr
+	variables := make(map[string]interface{})
+	enabled := false
+	var experimentKey, variationKey string
+
+	feature, _ := projectConfig.GetFeatureByKey(key)
+	variable := entities.Variable{}
+	decisionContext := decision.FeatureDecisionContext{
+		Feature:       &feature,
+		ProjectConfig: projectConfig,
+		Variable:      variable,
+	}
+
+	if featureDecision, err := c.DecisionService.GetFeatureDecision(decisionContext, uc); err == nil {
+		if featureDecision.Variation != nil {
+			enabled = featureDecision.Variation.FeatureEnabled
+			variationKey = featureDecision.Variation.Key
+			experimentKey = featureDecision.Experiment.Key
+
+			if featureDecision.Source == decision.FeatureTest && !disableTracking {
+				// send impression event for feature tests
+				impressionEvent := event.CreateImpressionUserEvent(decisionContext.ProjectConfig, featureDecision.Experiment, *featureDecision.Variation, uc)
+				c.EventProcessor.ProcessEvent(impressionEvent)
+			}
+		}
+
+		errs := new(multierror.Error)
+		for _, v := range feature.VariableMap {
+			val := v.DefaultValue
+			if enabled {
+				if variable, ok := featureDecision.Variation.Variables[v.ID]; ok {
+					val = variable.Value
+				}
+			}
+			value, e := c.GetTypedFeatureVariableValue(val, v)
```

Comment:
What is `GetTypedFeatureVariableValue`? I can't find this in the SDK, is this something added in another branch?

Reply:
https://github.com/optimizely/go-sdk/compare/yasir/OASIS-6247